src/library_workerfs.js

```diff
@@ -0,0 +1,95 @@
+mergeInto(LibraryManager.library, {
+  $WORKERFS__deps: ['$FS'],
+  $WORKERFS: {
+    DIR_MODE: {{{ cDefine('S_IFDIR') }}} | 511 /* 0777 */,
+    FILE_MODE: {{{ cDefine('S_IFREG') }}} | 511 /* 0777 */,
+    mount: function (mount) {
+      assert(ENVIRONMENT_IS_WORKER);
+      var root = WORKERFS.createNode(null, '/', WORKERFS.DIR_MODE, 0);
+      (mount.opts.files || []).forEach(function(file) {
```

Comment:
Oh, I see, so if a `File` is provided, we would have the name already? I think though that people might want a different name when adding it to the emscripten filesystem (the local file could be `/home/name/../../long-name` and the emscripten FS one might be `short-name`). Also I'm not sure the timestamp would be useful.

It seems simpler to me to just accept `Blobs`, unless I am missing an advantage here.